service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,47 @@
         </actions>
     </service>
 
+    <service verb="get" noun="PackageInfo">
+        <implements service="mantle.shipment.CarrierServices.get#AutoPackageInfo"/>
+        <actions>
+            <!--Get shipmentIds from ShipmentAndItemSource using orderId and orderItemSeqId of itemInfoList-->
+            <set field="shipmentIdsSet" from="new HashSet()"/>
+            <iterate list="itemInfoList" entry="itemInfoMap">
+                <entity-find entity-name="mantle.shipment.ShipmentAndItemSource" list="shipmentSourceList">
+                    <econdition field-name="orderId" from="itemInfoMap.orderId"/>
+                    <order-by field-name="orderItemSeqId"/>
+                </entity-find>
+               <set field="shipmentIdsSet" from="new HashSet(shipmentSourceList*.shipmentId)"/>
```

Comment:
This will over write the values every time in your loop. Please check.
Can you use operator="in" on orderId to get the shipmentSourceList, and then collect the shipmentIds?